h/static/scripts/base/controller.js

```diff
@@ -25,6 +25,23 @@ function findRefs(el) {
 }
 
 /**
+ * Dispatch an event to all event handlers of an element's controller registered
+ * using `Controller#on`
+ */
+function dispatchEvent(element, event) {
+  if (!element.controllers) {
+    return;
+  }
+  element.controllers.forEach(ctrl => {
+    ctrl._listeners.forEach(listener => {
```

Comment:
This code is now gone as I've removed the synthetic event handling for the moment. TBQH this is unlikely to make much of a difference but I'll remember it if we do decide we need something similar in future.